packages/plugin-types/src/deep_partial.ts

```diff
@@ -1,7 +1,11 @@
 /**
- * Makes all the properties of the entire nested object partial
- * instead of just the top level properties
+ * Makes all the properties of the entire nested object partial for Amplify generated configs
+ * instead of just the top level properties. Other properties are not changed.
  */
-export type DeepPartial<T> = {
-  [P in keyof T]?: DeepPartial<T[P]>;
+export type DeepPartialAmplifyGeneratedConfigs<T> = {
+  [P in keyof T]?: P extends 'auth' | 'data' | 'storage'
+    ? T[P] extends object
+      ? DeepPartialAmplifyGeneratedConfigs<T[P]>
+      : Partial<T[P]>
```

Comment:
if `T[P]` is not an object, does `Partial<>` have any effect? Should this just be `T[P]`?

Reply:
Yes, `Partial<>` will make the properties under `auth`, `data`, and `storage` optional. For example customers can add `data.api_key` to outputs without having to pass `aws_region` and other required fields that are generated by Amplify.

Reply:
FWIW, if `T[P]` is required, is an object, and under one of the 3 mentioned above, then this won't be affected by `Partial<>` and still be required. I haven't seen such that exists so this is not a problem currently but can be if we introduce something like this in the future.